wrapanapi/systems/google.py

```diff
@@ -0,0 +1,798 @@
+            if error.resp.status == 404:
+                raise VMInstanceNotFound(self._name)
+            else:
+                raise
+        return self.raw
+
+    def _get_state(self):
+        self.refresh()
+        return self._api_state_to_vmstate(self.raw['status'])
+
+    @property
+    def ip(self):
+        self.refresh()
+        return self.raw.get('networkInterfaces')[0].get('networkIP')
+
+    @property
+    def ip_external(self):
+        self.refresh()
+        access_configs = self.raw.get('networkInterfaces')[0].get('accessConfigs')[0]
+        return access_configs.get('natIP')
+
+    @property
+    def type(self):
+        if self.raw.get('machineType', None):
+            return self.raw['machineType'].split('/')[-1]
+        return None
+
+    @property
+    def creation_time(self):
+        creation_time = iso8601.parse_date(self.raw['creationTimestamp'])
+        return creation_time.astimezone(pytz.UTC)
+
+    def delete(self, timeout=250):
+        self.logger.info("Deleting Google Cloud instance '%s'", self.name)
+        operation = self._api.delete(
+            project=self._project, zone=self.zone, instance=self.name).execute()
+
+        wait_for(
+            lambda: self.system.is_zone_operation_done(operation['name']), delay=0.5,
+            num_sec=timeout, message="Delete {}".format(self.name)
+        )
+
+        self.logger.info(
+            "DELETE request successful, waiting for instance '%s' to be removed...",
+            self.name
+        )
+        wait_for(
+            lambda: not self.exists, delay=0.5, num_sec=timeout,
+            message=" instance '{}' to not exist".format(self.name)
+        )
+        return True
+
+    def cleanup(self):
+        return self.delete()
+
+    def restart(self):
+        self.logger.info("Restarting Google Cloud instance '%s'", self.name)
+        # Use self.stop/self.start vs reset since it's easier to block
+        # (the VM stays in RUNNING state when using reset)
+        return self.stop() and self.start()
+
+    def stop(self):
+        self.logger.info("Stopping Google Cloud instance '%s'", self.name)
+        operation = self._api.stop(
+            project=self._project, zone=self.zone, instance=self.name).execute()
+        wait_for(
+            lambda: self.system.is_zone_operation_done(operation['name']),
+            message="stop operation done {}".format(self.name), timeout=360
+        )
+        self.wait_for_state(VmState.STOPPED)
+        return True
+
+    def start(self):
+        self.logger.info("Starting Google Cloud instance '%s'", self.name)
+        operation = self._api.start(
+            project=self._project, zone=self.zone, instance=self.name).execute()
+        wait_for(
+            lambda: self.system.is_zone_operation_done(operation['name']),
+            message="start operation done {}".format(self.name)
+        )
+        self.wait_for_state(VmState.RUNNING)
+        return True
+
+
+class GCEImage(Template):
```

Comment:
**Optional**

I'm thinking we should stick with a consistent prefix for these classes, but that would mean we'd be stuck with `GoogleCloudImage` and `GoogleCloudInstance`.

Reply:
Changing to GoogleCloud<Entity>